app/assets/javascripts/app/form-field-format.js

```diff
@@ -35,6 +35,17 @@ function formatForm() {
       if (el !== '.mfa') document.activeElement.blur();
     }
   });
+
+  const submitBtnSel = 'input[type="submit"]';
+  const btnEl = document.querySelector(submitBtnSel);
+  const safeSubmit = function() {
+    btnEl.disabled = true;
+    document.querySelector('form').submit();
+  };
+  if (btnEl) {
+    btnEl.addEventListener('submit', safeSubmit, false);
+    btnEl.addEventListener('click', safeSubmit, false);
```

Comment:
Do we need both listeners?

Reply:
Technically you can do an enter and a click

Reply:
Double check me on this, but I think "submit" will always trigger. You may also be able to side-step it by listening for the form submit and intercepting that.

Reply:
You are correct. Nice catch! I was baffled why that didn't work in the first place. The reason it didn't work is because I had the submit event listener on the button instead of the form!  Anyway all this code is about to be replaced by the opt in and touching a lot more files.  I'm not sure how I can write specs for this though. Thoughts?  I'm definitely not seeing double posts on the forms.

Reply:
spoke with moncef and he was ok with just testing that the button is disabled